File: 3616-make-array-elements-equal-to-zero/make-array-elements-equal-to-zero.py
class Solution:
    def countValidSelections(self, nums: List[int]) -> int:
        tot = sum(nums)
        cur = 0
        res = 0
        for i in range(len(nums)):
            if nums[i] == 0 and cur == tot-cur:
                res += 2
            elif nums[i] == 0 and abs(cur-(tot-cur)) <= 1:
                res += 1
            cur += nums[i]
        return res

File: 3616-make-array-elements-equal-to-zero/make-array-elements-equal-to-zero.cpp
class Solution {
public:
    int countValidSelections(vector<int>& nums) {
        int tot = 0, res = 0, leftSum = 0;
        for (int num: nums) tot += num;
        for (int num: nums){
            if (num == 0){
                if (leftSum == tot-leftSum) res += 2;
                else if (std::abs(leftSum - (tot-leftSum)) == 1) res++;
            }
            leftSum += num;
        }
        return res;
    }
};